src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -138,80 +189,184 @@ public Void call() throws Exception {
-        }).getId().toString();
+        }).getId();
     }
 
 
     @Override
     public String updateDeviceMember(ManagerDeviceMemberElement deviceElement, String name, String vserverIpAddress,
             String contactIpAddress, String ipAddress, String gateway, String prefixLength) throws Exception {
-        return deviceElement.getId();
+
+        DeviceMemberEntity deviceMember = new DeviceMemberEntity();
+        deviceMember.setId(Long.parseLong(deviceElement.getId()));
+        deviceMember.setName(name);
+        return updateDeviceMember(deviceMember);
+        // TODO - Sudhir Add other information like gateway, ipAddress
+    }
+
+    public String updateDeviceMember(DeviceMemberEntity deviceElement) throws Exception {
+
+        return this.txControl.required(new Callable<DeviceMemberEntity>() {
+
+            @Override
+            public DeviceMemberEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+
+                CriteriaQuery<DeviceMemberEntity> query = criteriaBuilder.createQuery(DeviceMemberEntity.class);
+                Root<DeviceMemberEntity> r = query.from(DeviceMemberEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("id"), deviceElement.getId())));
+
+                List<DeviceMemberEntity> result = IsmDeviceApi.this.em.createQuery(query).getResultList();
+                if (result.isEmpty()) {
+                    String msg = String.format("Cannot find the Device Member Entity " + "Id: %s\n",
+                            deviceElement.getId());
+                    LOG.error(msg);
+                    throw new IllegalArgumentException(msg);
+                }
+                result.get(0).setName(deviceElement.getName());
+                IsmDeviceApi.this.em.persist(result.get(0));
+                return result.get(0);
+            }
+        }).getId();
     }
 
     @Override
     public void deleteDeviceMember(String id) throws Exception {
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DeviceMemberEntity memberResult = IsmDeviceApi.this.em.find(DeviceMemberEntity.class,
+                        Long.parseLong(id));
+                if (memberResult == null) {
+                    LOG.info("Attempt to delete device member id:...." + Long.parseLong(id));
+                    LOG.info("..device member not found, no-op.");
+                    return null;
+                }
+                IsmDeviceApi.this.em.remove(memberResult);
+                return null;
+            }
+        });
     }
 
     @Override
     public ManagerDeviceMemberElement getDeviceMemberById(final String id) throws Exception {
-        return txControl.supports(new Callable<ManagerDeviceMemberElement>() {
-            
+        return this.txControl.supports(new Callable<DeviceMemberEntity>() {
```

Comment:
Again use the vsid also in the case to find the parent. Also observe the symmetry of this approach, when you create a device member you are using/expecting/enforcing the parent. When retrieving/deleting/updating you should also do the same. The lack of symmetry in this case is odd and incorrect.